app/code/Magento/InventoryCatalog/etc/di.xml

```diff
@@ -8,4 +8,7 @@
 <config xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xsi:noNamespaceSchemaLocation="urn:magento:framework:ObjectManager/etc/config.xsd">
     <preference for="Magento\InventoryCatalog\Api\DefaultStockProviderInterface" type="Magento\InventoryCatalog\Model\DefaultStockProvider"/>
     <preference for="Magento\InventoryCatalog\Api\DefaultSourceProviderInterface" type="Magento\InventoryCatalog\Model\DefaultSourceProvider"/>
+    <type name="Magento\Inventory\Model\StockRepository">
```

Comment:
We need to add this plugin not on implementation `Magento\Inventory\Model\StockRepository` but on interface `\Magento\InventoryApi\Api\StockRepositoryInterface`

If somebody will replace `StockRepository` on own implementation our check will be work